src/js/views/maps/legend/CategoricalSwatchView.js

```diff
@@ -52,6 +66,59 @@ define([
           "background-color",
           this.model.getCss(),
         );
+
+        if (this.filterModel) {
+          // Add layer-item class to the view's main element
+          // This class is added to reuse functionality of the class from layer list items
+          this.el.classList.add("layer-item");
+
+          // Rebind event after DOM content is replaced
+          this.delegateEvents();
```

Comment:
No need to re-delegate. Backbone does this during render automatically. Unless this fixed a problem that I didn't come across in testing?